python/ray/experimental/data/datasource/file_based_datasource.py

```diff
@@ -158,7 +157,9 @@ def _resolve_paths_and_filesystem(
             filesystems inferred from the provided paths to ensure
             compatibility.
     """
-    from pyarrow.fs import FileType, _resolve_filesystem_and_path
+    from pyarrow.fs import FileSystem, FileType, \
+        PyFileSystem, FSSpecHandler, \
+        _resolve_filesystem_and_path
```

Comment:
Should we be using this private method from pyarrow.fs?

Reply:
I think in the long term, probably not. 

@clarkzinzow do you know why we need this?

Reply:
Agreed on not using it in the long-run; I was manually implementing path + filesystem resolution when I found that helper and decided to use it instead of duplicating the logic.